### DISCUSSION TITLE
Adding smooth animation on custom hover class

### DISCUSSION THREAD

**Discussion by mahendraAtMave** [tailwindlabs/tailwindcss#12925] [Help] [answered]
so I'm adding a custom linear gradient bg on hover on a card but I want to add smooth animation on hover, I tried adding `transition` and `delay` properties but it's not working.
here's my card code:

```javascript 
     <div
      className={cn(
        "flex flex-col items-center gap-2 bg-[#E0EFF5] px-10 pt-6 transition delay-500 duration-300 hover:rounded-md hover:bg-[linear-gradient(180deg,_#E0EFF5_26.89%,_#65AFCD__80.77%)]",
        className,
      )}
    >
      {/* heading and subHeading */}
      <div className="flex flex-col items-center justify-center gap-1">
        <span className="text-xs font-medium uppercase tracking-[0.075rem] text-[#65AFCD]">
          therapy for me
        </span>
        <h3 className="text-[2.5rem] font-semibold leading-[3rem] tracking-[-0.05rem] text-[#3F9CC0]">
          individual
        </h3>
      </div>

      {/* Image */}
      <div className="relative h-60 w-56">
        <Image fill src={therapyCardImage1} alt="model image" />
      </div>
    </div>

**Comment by mahendraAtMave** [comment]
@wongjn any help is appreciated :)

**Reply by wongjn** [reply, marked as answer]
You can't transition between `none` and a `linear-gradient()` for `background-image`. Instead, you could consider having a separate element with the `linear-gradient()` that you transition in with `opacity`:

```diff
 <div
   className={cn(
-    "flex flex-col items-center gap-2 bg-[#E0EFF5] px-10 pt-6 transition delay-500 duration-300 hover:rounded-md hover:bg-[linear-gradient(180deg,_#E0EFF5_26.89%,_#65AFCD__80.77%)]",
+    "flex flex-col items-center gap-2 bg-[#E0EFF5] px-10 pt-6 hover:rounded-md relative z-0 group overflow-hidden",
     className,
   )}
 >
+   <div className="absolute inset-0 -z-10 opacity-0 transition delay-500 duration-300 bg-[linear-gradient(180deg,_#E0EFF5_26.89%,_#65AFCD__80.77%)] group-hover:opacity-100"></div>
    {/* heading and subHeading */}
    <div className="flex flex-col items-center justify-center gap-1">
```

[See this Tailwind Play](https://play.tailwindcss.com/hUULH0i78K) for a working example.

**Reply by mahendraAtMave** [reply]
I got it, but I lost my border-radius animation now, why is that? like I'm adding border-radius on hover @wongjn

**Reply by wongjn** [reply]
I'm not sure what you mean when you say you "lost your border-radius animation" – there would not be any `border-radius` animation from your original code.

However, to *add* a `border-radius` transition animation, you can apply to the top-level `<div>`:
- A positive duration `transition-duration` via a `duration-*` class.
- A `transition-*` class that includes `border-radius` in the `transition-property` value, like:
  - `transition-property: all` from `transition-all`
  - `transition-property: border-radius` from `transition-[border-radius]`

See these suggestions working in [this Tailwind Play](https://play.tailwindcss.com/ujyTSxb4E1).

**Reply by mahendraAtMave** [reply]
got it, really appreciate your help mate 🙌🏻
